Babylon/groups/azure/groups/acr/commands/pull.py

```diff
@@ -1,44 +1,52 @@
-import os
+import subprocess
 import logging
 import docker
-from click import command, pass_context
+from click import command, pass_context, option
 from click.core import Context
 from azure.containerregistry import ContainerRegistryClient
-from Babylon.utils.decorators import requires_external_program, require_deployment_key
+from Babylon.utils.decorators import require_deployment_key
 
 logger = logging.getLogger("Babylon")
 
 """Command Tests
-> babylon acr pull
-Should provide a clean error log
-> babylon acr pull
-Should provide a clean error log
-> babylon acr pull
+> babylon azure acr pull
 Should add a new entry to `docker image ls`
-> babylon acr pull
-Should a new entry to `docker image ls` with the latest tag
+> babylon azure acr pull my_image:latest
+Should add a new entry to `docker image ls`
+> babylon azure acr pull -r my_registry my_image:my_tag
+Should add new entry to `docker image ls`
 """
 
+
 @command()
-@requires_external_program("docker")
 @pass_context
 @require_deployment_key("acr_src_registry_name", "acr_src_registry_name")
 @require_deployment_key("acr_image_reference", "acr_image_reference")
-def pull(ctx: Context, acr_src_registry_name: str, acr_image_reference: str):
+@option("-i", "--image")
+@option("-r", "--registry")
+def pull(ctx: Context, acr_src_registry_name: str, acr_image_reference: str, registry: str, image: str):
```

Comment:
For values that can be optional (mostly parameters tied to `@option` we should use the type `typing.Optional[str]`